source/Halibut/DataStream.cs

```diff
@@ -9,18 +11,38 @@ namespace Halibut
     public class DataStream : IEquatable<DataStream>, IDataStreamInternal
     {
         readonly Action<Stream> writer;
+        readonly Func<Stream, CancellationToken, Task> writerAsync;
         IDataStreamReceiver receiver;
 
         [JsonConstructor]
         public DataStream()
         {
         }
-
-        public DataStream(long length, Action<Stream> writer)
+        
+        public DataStream(long length, Action<Stream> writer, Func<Stream, CancellationToken, Task> writerAsync)
```

Comment:
Could we make this internal to discourage normal users from giving both?

Reply:
We probably do want to give both in `Octopus` while we roll out.